.github/workflows/e2e.yml

```diff
@@ -0,0 +1,111 @@
+name: E2E
+
+permissions:
+  contents: read
+
+on:
+  workflow_dispatch:
+  push:
+    branches: ["main"]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: ["main"]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  e2e-tests:
+    name: E2E tests
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        k8s-name:
+          - k8s-oldest
+          - k8s-plus-one
+
+        dashboard-mode:
+          - read-only
+          - read-write
+
+        # k8s-name above is used to give stable naming of the checks for branch
+        # protection config. Map name to corresponding version for use in steps
+        include:
+          - k8s-name: k8s-oldest
+            k8s-version: v1.29.x
+          - k8s-name: k8s-plus-one
+            k8s-version: v1.30.x
+
+    env:
+      GOPATH: ${{ github.workspace }}
+      GO111MODULE: on
+      KO_DOCKER_REPO: registry.local:5000/tekton
+      CLUSTER_DOMAIN: c${{ github.run_id }}.local
+      ARTIFACTS: ${{ github.workspace }}/artifacts
+
+    steps:
+      - name: Harden runner
+        uses: step-security/harden-runner@0080882f6c36860b6ba35c610c98ce87d4e2f26f # v2.10.2
+        with:
+          egress-policy: audit
+
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          path: ${{ github.workspace }}/src/github.com/tektoncd/dashboard
+
+      - name: Checkout setup-kind.sh
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          repository: tektoncd/pipeline
+          ref: d306d649df2dbd2badaba6a90459efd05c753d2f
```

Comment:
Referencing the setup-kind.sh script from tektoncd/pipeline for now since it's a slightly modified version of the one in tektoncd/plumbing to work with GitHub Actions by adding `sudo` when updating the hosts file.

Longer term we'll have a reusable action that can be shared across the projects.